Source/WebCore/accessibility/AccessibilityNodeObject.cpp

```diff
@@ -2317,22 +2317,16 @@ static bool shouldPrependSpace(AXCoreObject& object, AXCoreObject* previousObjec
 
 String AccessibilityNodeObject::textUnderElement(TextUnderElementMode mode) const
 {
-    auto isAriaVisible = [this] () {
-        return Accessibility::findAncestor<AccessibilityObject>(*this, true, [] (const auto& object) {
-            return equalLettersIgnoringASCIICase(object.getAttribute(aria_hiddenAttr), "false"_s);
-        }) != nullptr;
-    };
-
     RefPtr node = this->node();
     if (auto* text = dynamicDowncast<Text>(node.get()))
-        return !mode.isHidden() || isAriaVisible() ? text->wholeText() : emptyString();
+        return !mode.isHidden() && !isAXHidden() ? text->wholeText() : emptyString();
```

Comment:
I built this patch and removed the `&& !isAXHidden()` check you added here, and no tests failed. Do we actually need this?